### PR TITLE
[Friesian] Friesian Helm chart release

### PR DESCRIPTION
**Pull request by hoshibara**
### 1. Why the change?

The AI-workflow team needs the Friesian server helm chart as a dependency of their repo.
This helm chart allows users to easily deploy the Friesian recommendation system on the k8s cluster.

### 2. User API changes

Refer to the readme document.

### 3. Summary of the change 

* Release Friesian helm chart

### 4. How to test?
- [x] Application test

### CODE REVIEW

File: apps/friesian-server-helm/friesian-helm/values.yaml [hunk 1]
```diff
@@ -0,0 +1,685 @@
...
+## define the resource PVC name
+resourcePVCName: friesian-serving-resources
+
+init:
+  ## enable the init step, include the feature, feature-recall, and recall server
+  enabled: true
+
+  ## Image for init steps
+  image:
+    repository: 10.239.45.10/arda/intelanalytics/friesian-serving
```
Comment (cyita) [inline apps/friesian-server-helm/friesian-helm/values.yaml:20]: Please use the docker image on docker hub.

Comment (cyita) [reply apps/friesian-server-helm/friesian-helm/values.yaml:20]: intelanalytics/friesian-serving:2.2.0-SNAPSHOT

File: apps/friesian-server-helm/readme.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Friesian server deploying with Helm Chart
+
+This example demonstrates how to deploy a friesian recommender server with Helm Chart.
```
Comment (hkvision) [inline apps/friesian-server-helm/readme.md:3]: friesian -> Friesian
Add on a kubernetes cluster?

File: apps/friesian-server-helm/readme.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+## Prepare environments
+
+* AVX512 instruction support in nodes deploying the recall server
```
Comment (hkvision) [inline apps/friesian-server-helm/readme.md:7]: in -> on
deploying -> that deploy

File: apps/friesian-server-helm/readme.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+* [Preparation steps](./preparation) or a PV contains resource files
+* [Helm](https://helm.sh) 3.2.0+
+* Kubernetes 1.19+
+
+## Deploy friesian server
+
+You can get the full installation process from [here](./friesian-helm/readme.md).
+
+## TL;DR
```
Comment (hkvision) [inline apps/friesian-server-helm/readme.md:16]: what does this mean?

File: apps/friesian-server-helm/readme.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+```bash
+git clone https://github.com/intel-analytics/BigDL.git
+cd BigDL/apps/friesian-server-helm
+
+# create a namespace
+kubectl create ns friesian
+
+# create a PVC to store resources and a pod to copy resources
+kubectl apply -f './preparation/1. create-pvc.yaml' -n friesian
```
Comment (hkvision) [inline apps/friesian-server-helm/readme.md:26]: can we avoid the space after 1. and 2.?

File: apps/friesian-server-helm/readme.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+# delete pod no longer used
+kubectl delete -f '2. create-volume-pod.yaml' -n friesian
+
+# install friesian server in namespace 'friesian' with name 'my-release'
+helm upgrade --install --debug -n friesian my-release ./friesian-helm
+
+```
+
+After installation, you can follow the printed information to check whether the servers running
```
Comment (hkvision) [inline apps/friesian-server-helm/readme.md:40]: running -> are running

File: apps/friesian-server-helm/readme.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+kubectl apply -f './preparation/1. create-pvc.yaml' -n friesian
+kubectl apply -f './preparation/2. create-volume-pod.yaml' -n friesian
+
+# copy resources to PVC
+kubectl cp /path/to/your/resources friesian/volume-pod:/resources
+
+# delete pod no longer used
+kubectl delete -f '2. create-volume-pod.yaml' -n friesian
+
+# install friesian server in namespace 'friesian' with name 'my-release'
```
Comment (hkvision) [inline apps/friesian-server-helm/readme.md:35]: any better name for my-release?

Comment (hoshibara) [reply apps/friesian-server-helm/readme.md:35]: my-release is used as the release name, which is common in other helm charts. 
I think its sense of violation can make users better notice this parameter.

Comment (hkvision) [reply apps/friesian-server-helm/readme.md:36]: okay

File: apps/friesian-server-helm/preparation/readme.md [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+
+1. Feature Server
+
+   Users' [WND](https://github.com/intel-analytics/BigDL/tree/main/python/friesian/example/wnd)
+   feature parquet file.
+
+   Items' [WND](https://github.com/intel-analytics/BigDL/tree/main/python/friesian/example/wnd)
+   feature parquet file.
+
+   (Skip this step if having initialized Redis)
```
Comment (hkvision) [inline apps/friesian-server-helm/preparation/readme.md:19]: if you have

File: apps/friesian-server-helm/preparation/readme.md [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+
+2. Feature-Recall Server
+
+   Users' embedding parquet file. (Skip this step if having initialized Redis)
+
+3. Recall Server
+
+   Items' embedding parquet file. (Skip this step if having initialized faiss model)
```
Comment (hkvision) [inline apps/friesian-server-helm/preparation/readme.md:27]: same as above

File: apps/friesian-server-helm/preparation/readme.md [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+   feature parquet file.
+
+   Items' [WND](https://github.com/intel-analytics/BigDL/tree/main/python/friesian/example/wnd)
+   feature parquet file.
+
+   (Skip this step if having initialized Redis)
+
+2. Feature-Recall Server
+
+   Users' embedding parquet file. (Skip this step if having initialized Redis)
```
Comment (hkvision) [inline apps/friesian-server-helm/preparation/readme.md:23]: embedding parquet file from two tower model? add link to two tower

File: apps/friesian-server-helm/preparation/readme.md (hunk 9 above)
Comment (hkvision) [reply apps/friesian-server-helm/preparation/readme.md:27]: model -> index

File: apps/friesian-server-helm/preparation/readme.md [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+2. Claim a [PV](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) to store resource
+   files. Edit `create-pvc.yaml` file as follows:
+
+    * `spec.resources.requests.storage`: Size of the persistent volume, large enough to contain the
+      resource files and the faiss model generated during initialization.
+    * `spec.storageClassName`: Name of the available
+      cluster [storage class](https://kubernetes.io/docs/concepts/storage/storage-classes/). If
+      using AWS, Alibaba Cloud, or other cloud services, we can obtain this name in the
+      dashboard/console.
+
```
Comment (hkvision) [inline apps/friesian-server-helm/preparation/readme.md:52]: Add a sentence here to instruct users to run the command?

File: apps/friesian-server-helm/preparation/readme.md [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+    kubectl get pv -n friesian
+    ```
+
+3. Create a pod to attach the claimed PV.
+
+    ```bash
+    kubectl apply -f 'create-volume-pod.yaml' -n friesian
+    ```
+
+4. Copy resource files into the attached PV and check files' path.
```
Comment (hkvision) [inline apps/friesian-server-helm/preparation/readme.md:66]: the resouce ..
check the path of the files

File: apps/friesian-server-helm/preparation/readme.md [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+    |-part-xxxxxxxxx-xxxx.parquet
+    |-part-xxxxxxxxx-xxxx.parquet
+    |-...
+  |-wnd_item.parquet (Users' feature parquet file) 
+    |                (Fill wnd_item.parquet in `init.feature.resourcePath.initialItemDataPath`)
+    |-part-xxxxxxxxx-xxxx.parquet
+    |-part-xxxxxxxxx-xxxx.parquet
+    |-...
+  | (Feature-Recall Server)
+  |-tmpDir
```
Comment (hkvision) [inline apps/friesian-server-helm/preparation/readme.md:100]: what is tmpDir?

File: apps/friesian-server-helm/friesian-helm/readme.md [hunk 14]
```diff
@@ -0,0 +1,98 @@
+# Friesian Online Serving Helm Chart
+
+This document demonstrates how to use the helm chart to deploy the Friesian online serving pipeline
```
Comment (hkvision) [inline apps/friesian-server-helm/friesian-helm/readme.md:3]: move the link of helm from line10 to here

File: apps/friesian-server-helm/friesian-helm/readme.md [hunk 15]
```diff
@@ -0,0 +1,98 @@
...
+## Prepare environments
+
+* AVX512 instruction support on nodes that deploy the recall server
+* [Preparation steps](../preparation) or a PV contains resource files
+* [Helm](https://helm.sh) 3.2.0+
+* Kubernetes 1.19+
+
+## Modify the configurations
+
+In [values.yaml](./values.yaml), you can edit parameters to match your cluster config.
```
Comment (hkvision) [inline apps/friesian-server-helm/friesian-helm/readme.md:15]: edit the parameters according to your cluster config

File: apps/friesian-server-helm/friesian-helm/readme.md [hunk 16]
```diff
@@ -0,0 +1,98 @@
...
+| `prometheus`                   | [Prometheus stack](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack) that serviceMonitors depend on. <br />Can be disabled when external prometheus stack is deployed. <br /> Nested components refer to [prometheus chart readme](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack) | `prometheus.enabled`                   |
+| `feature`                      | Feature server supply features                                                                                                                                                                                                                                                                                                                                           | Always On                              |
```
Comment (hkvision) [inline apps/friesian-server-helm/friesian-helm/readme.md:51]: Feature server for ranking model features

File: apps/friesian-server-helm/friesian-helm/readme.md [hunk 17]
```diff
@@ -0,0 +1,98 @@
...
+| `feature.serviceMonitor`       | Service Monitor for `feature`. <br />**`prometheus` must be enabled (or installed)**                                                                                                                                                                                                                                                                                     | `feature.serviceMonitor.enabled`       |
+| `featureRecall`                | Feature server supply user embedding                                                                                                                                                                                                                                                                                                                                     | Always On                              |
```
Comment (hkvision) [inline apps/friesian-server-helm/friesian-helm/readme.md:53]: for user embedding

File: apps/friesian-server-helm/friesian-helm/readme.md [hunk 18]
```diff
@@ -0,0 +1,98 @@
...
+
+The Helm chart
+uses [kube-prometheus-stack](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack)
+to monitor cluster.
+This stack creates a Service for scraping kubelet information (defined
+in `kube-prometheus-stack.prometheusOperator.kubeletService`) which isn't deleted when uninstalling.
+See [this issue](https://github.com/SumoLogic/sumologic-kubernetes-collection/issues/1101) for a
+detailed explanation.
+
+To remove this service after uninstalling, run:
```
Comment (hkvision) [inline apps/friesian-server-helm/friesian-helm/readme.md:89]: uninstallation

File: apps/friesian-server-helm/friesian-helm/readme.md [hunk 19]
```diff
@@ -0,0 +1,98 @@
...
+
+```bash
+helm delete my-release -n friesian
+```
+
+The Helm chart
+uses [kube-prometheus-stack](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack)
+to monitor cluster.
+This stack creates a Service for scraping kubelet information (defined
+in `kube-prometheus-stack.prometheusOperator.kubeletService`) which isn't deleted when uninstalling.
```
Comment (hkvision) [inline apps/friesian-server-helm/friesian-helm/readme.md:85]: when uninstalling -> after uninstallation

File: apps/friesian-server-helm/friesian-helm/readme.md [hunk 20]
```diff
@@ -0,0 +1,98 @@
...
+To remove this service after uninstalling, run:
+
+```bash
+kubectl delete svc <release_name>-prometheus-kubelet -n kube-system
+
+# In this demo, replace <release_name> with my-release
+kubectl delete svc my-release-prometheus-kubelet -n kube-system
+```
+
+The commands remove all the Kubernetes components associated with the chart and delete the release.
```
Comment (hkvision) [inline apps/friesian-server-helm/friesian-helm/readme.md:98]: Make this ending sentence clearer.
The commands mean all the commands in this section or the commands only in the previous bash segment?

**Comment by hkvision** [comment]
rename readme.md to README.md to align with other places in BigDL.

File: apps/friesian-server-helm/README.md [hunk 21]
```diff
@@ -0,0 +1,42 @@
...
+* AVX512 instruction support on nodes that deploy the recall server
+* [Preparation steps](./preparation) or a PV contains resource files
+* [Helm](https://helm.sh) 3.2.0+
+* Kubernetes 1.19+
+
+## Deploy Friesian server
+
+You can get the full installation document from [here](./friesian-helm/README.md).
+
+## TL;DR
```
Comment (cyita) [inline apps/friesian-server-helm/README.md:17]: May be use a formal tile instead of TL;DR?

Comment (hkvision) [reply apps/friesian-server-helm/README.md:17]: agree

File: apps/friesian-server-helm/README.md [hunk 22]
```diff
@@ -0,0 +1,42 @@
+# Friesian server deploying with Helm Chart
```
Comment (cyita) [inline apps/friesian-server-helm/README.md:1]: Friesian online serving pipeline

File: apps/friesian-server-helm/friesian-helm/README.md [hunk 23]
```diff
@@ -0,0 +1,98 @@
...
+    * [`init.recall.config`](./values.yaml#L81)
+    * [`feature.config`](./values.yaml#L114)
+    * [`featureRecall.config`](./values.yaml#L196)
+    * [`recall.config`](./values.yaml#L275)
+    * [`recall.resourcePath`](./values.yaml#L282)
+    * [`ranking.config`](./values.yaml#L330)
+    * [`ranking.resourcePath`](./values.yaml#L339)
+    * [`recommender.config`](./values.yaml#L385)
+
+* Modify all image repositories & tags to match your docker-hub.
```
Comment (cyita) [inline apps/friesian-server-helm/friesian-helm/README.md:36]: -> to match your docker images.

**Review by hkvision — APPROVED** [review]

File: apps/friesian-server-helm/README.md (hunk 21 above)
Comment (hoshibara) [reply apps/friesian-server-helm/README.md:17]: modified to 'Quick Installation'

**Pull request merged**